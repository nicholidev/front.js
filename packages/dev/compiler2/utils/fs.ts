import { promises as fsp } from "fs";
import * as path from "path";

export async function writeFileSafe(
  file: string,
  contents: string
): Promise<string> {
  await fsp.mkdir(path.dirname(file), { recursive: true });
  await fsp.writeFile(file, contents);
  return file;
}

export async function writeFilesSafe(
  files: { file: string; contents: string }[]
): Promise<string[]> {
  return Promise.all(
    files.map(({ file, contents }) => writeFileSafe(file, contents))
  );
}

export async function createTemporaryDirectory(
  baseDir: string
): Promise<string> {
  return fsp.mkdtemp(path.join(baseDir, "remix-"));
}
