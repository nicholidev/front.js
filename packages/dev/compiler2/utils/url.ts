import * as path from "path";

export function createUrl(publicPath: string, file: string): string {
  return publicPath + file.split(path.win32.sep).join("/");
}
