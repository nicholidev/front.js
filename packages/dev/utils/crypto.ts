import type { BinaryLike } from "crypto";
import { createHash } from "crypto";

export function getHash(source: BinaryLike): string {
  return createHash("sha1").update(source).digest("hex");
}
