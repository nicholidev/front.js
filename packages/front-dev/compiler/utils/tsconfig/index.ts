import tsConfigPaths from "tsconfig-paths";

export function createMatchPath(tsconfigPath: string | undefined) {
  // There is no tsconfig to match paths against.
  if (!tsconfigPath) {
    return undefined;
  }

  // When passing a absolute path, loadConfig assumes that the path contains
  // a tsconfig file.
  // Ref.: https://github.com/dividab/tsconfig-paths/blob/v4.0.0/src/__tests__/config-loader.test.ts#L74
  let configLoaderResult = tsConfigPaths.loadConfig(tsconfigPath);

  if (configLoaderResult.resultType === "failed") {
    if (configLoaderResult.message === "Missing baseUrl in compilerOptions") {
      throw new Error(
        `ðŸš¨ Oops! No baseUrl found, please set compilerOptions.baseUrl in your tsconfig or jsconfig`
      );
    }
    return undefined;
  }

  return tsConfigPaths.createMatchPath(
    configLoaderResult.absoluteBaseUrl,
    configLoaderResult.paths,
    configLoaderResult.mainFields,
    configLoaderResult.addMatchAll
  );
}
