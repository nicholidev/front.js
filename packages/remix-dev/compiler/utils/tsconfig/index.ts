import tsConfigPaths from "tsconfig-paths";

import { loadTsConfig } from "./configLoader";
export { loadTsConfig } from "./configLoader";

export function createMatchPath() {
  let configLoaderResult = loadTsConfig();
  if (configLoaderResult.resultType === "failed") {
    return undefined;
  }

  let matchPath = tsConfigPaths.createMatchPath(
    configLoaderResult.absoluteBaseUrl,
    configLoaderResult.paths,
    configLoaderResult.mainFields,
    configLoaderResult.addMatchAll
  );

  return matchPath;
}
