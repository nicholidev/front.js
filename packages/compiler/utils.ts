import fs from "fs";
import path from "path";

export function readVendorFile(filename: string): string {
  let file = path.resolve(__dirname, "./vendor", filename);
  return fs.readFileSync(file, "utf-8");
}
