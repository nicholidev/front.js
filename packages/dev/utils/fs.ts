import { promises as fsp } from "fs";
import * as path from "path";
// @ts-expect-error
import readPackageJson from "read-package-json-fast";

export async function writeFileSafe(
  file: string,
  contents: string
): Promise<void> {
  await fsp.mkdir(path.dirname(file), { recursive: true });
  await fsp.writeFile(file, contents);
}

export async function createTemporaryDirectory(
  baseDir: string
): Promise<string> {
  return fsp.mkdtemp(path.join(baseDir, "remix-"));
}
