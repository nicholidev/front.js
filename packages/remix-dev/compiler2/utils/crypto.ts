import * as fs from "fs";
import type { BinaryLike } from "crypto";
import { createHash } from "crypto";

export function getHash(source: BinaryLike): string {
  return createHash("sha256").update(source).digest("hex");
}

export async function getFileHash(file: string): Promise<string> {
  return new Promise((accept, reject) => {
    let hash = createHash("sha256");
    fs.createReadStream(file)
      .on("error", error => reject(error))
      .on("data", data => hash.update(data))
      .on("close", () => {
        accept(hash.digest("hex"));
      });
  });
}
