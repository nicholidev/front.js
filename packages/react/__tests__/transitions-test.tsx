import type { Page } from "puppeteer";
import puppeteer from "puppeteer";

import { prettyHtml } from "../../core/__tests__/utils";

function reactIsHydrated(page: Page) {
  return page.waitForFunction("window.reactIsHydrated === true");
}

describe("route transitions", () => {
  describe("transitioning to a new route with data", () => {
    it("works", async () => {
      let browser = await puppeteer.launch();
      let page = await browser.newPage();
      await page.goto("http://localhost:3000");
      await reactIsHydrated(page);
      let content = await page.content();

      expect(prettyHtml(content)).toMatchSnapshot();

      await page.click('a[href="/gists"]');
      await page.waitForSelector('[data-test-id="/gists/index"]');
      content = await page.content();

      expect(prettyHtml(content)).toMatchSnapshot();

      return browser.close();
    });
  });

  describe("transitioning back after a reload", () => {
    it("works", async () => {
      let browser = await puppeteer.launch();
      let page = await browser.newPage();
      await page.goto("http://localhost:3000/gists");
      await reactIsHydrated(page);
      let content = await page.content();

      expect(prettyHtml(content)).toMatchSnapshot();

      await page.click('a[href="/gists/mjackson"]');
      await page.waitForSelector('[data-test-id="/gists/$username"]');
      content = await page.content();

      expect(prettyHtml(content)).toMatchSnapshot();

      await page.reload();
      await reactIsHydrated(page);

      await page.goBack();
      await page.waitForSelector('[data-test-id="/gists"]');
      content = await page.content();

      expect(prettyHtml(content)).toMatchSnapshot();
    });
  });
});
