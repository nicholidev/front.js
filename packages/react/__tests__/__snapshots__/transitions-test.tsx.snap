// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`route transitions transitioning back after a reload works 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <!--$-->
    <title>Public Gists</title>
    <meta name=\\"description\\" content=\\"View the latest gists from the public\\" />
    <!--/$-->
    <link
      rel=\\"stylesheet\\"
      href=\\"//unpkg.com/@exampledev/new.css@1.1.3/new.css\\"
    />
    <link rel=\\"stylesheet\\" href=\\"/build/assets/global-ec887178.css\\" />
    <link
      rel=\\"stylesheet\\"
      href=\\"/build/assets/style/routes/gists-d45b2a57.css\\"
    />
  </head>
  <body class=\\"m-4\\">
    <!--$-->
    <div data-test-id=\\"/gists\\">
      <header>
        <h1>Gists</h1>
        <ul>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/gists/ryanflorence\\"
              >Ryan Florence<!-- -->
            </a>
          </li>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/gists/mjackson\\"
              >Michael Jackson<!-- -->
            </a>
          </li>
        </ul>
      </header>
      <div data-test-id=\\"/gists/index\\">
        <h2>Public Gists</h2>
        <ul>
          <li><a>remix-server.jsx</a></li>
        </ul>
      </div>
    </div>
    <!--/$-->
    <script>
      __remixContext = {
        browserManifest: {
          \\"entry-browser\\": {
            fileName: \\"entry-browser.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
          \\"global.css\\": { fileName: \\"assets/global-ec887178.css\\" },
          \\"routes/gists\\": {
            fileName: \\"routes/gists.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
          \\"routes/gists/index\\": {
            fileName: \\"routes/gists/index.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
          \\"style/routes/gists.css\\": {
            fileName: \\"assets/style/routes/gists-d45b2a57.css\\",
          },
        },
        publicPath: \\"/build/\\",
        routeManifest: {
          \\"routes/gists\\": { id: \\"routes/gists\\", path: \\"gists\\" },
          \\"routes/gists/index\\": {
            id: \\"routes/gists/index\\",
            path: \\"/\\",
            parentId: \\"routes/gists\\",
          },
        },
        routeData: {
          \\"routes/gists\\": {
            users: [
              { id: \\"ryanflorence\\", name: \\"Ryan Florence\\" },
              { id: \\"mjackson\\", name: \\"Michael Jackson\\" },
            ],
          },
          \\"routes/gists/index\\": [
            {
              url:
                \\"https://api.github.com/gists/610613b54e5b34f8122d1ba4a3da21a9\\",
              id: \\"610613b54e5b34f8122d1ba4a3da21a9\\",
              files: { \\"remix-server.jsx\\": { filename: \\"remix-server.jsx\\" } },
              owner: {
                login: \\"ryanflorence\\",
                id: 100200,
                avatar_url:
                  \\"https://avatars0.githubusercontent.com/u/100200?v=4\\",
              },
            },
          ],
        },
        routeParams: { \\"routes/gists\\": {}, \\"routes/gists/index\\": {} },
      };
    </script>
    <script type=\\"module\\" src=\\"/build/entry-browser.js\\"></script>
  </body>
</html>
"
`;

exports[`route transitions transitioning back after a reload works 2`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <!--$-->
    <title>1 gists from mjackson</title>
    <meta name=\\"description\\" content=\\"View all of the gists from mjackson\\" />
    <!--/$-->
    <link
      rel=\\"stylesheet\\"
      href=\\"//unpkg.com/@exampledev/new.css@1.1.3/new.css\\"
    />
    <link rel=\\"stylesheet\\" href=\\"/build/assets/global-ec887178.css\\" />
    <link
      rel=\\"stylesheet\\"
      href=\\"/build/assets/style/routes/gists-d45b2a57.css\\"
    />
  </head>
  <body class=\\"m-4\\">
    <!--$-->
    <div data-test-id=\\"/gists\\">
      <header>
        <h1>Gists</h1>
        <ul>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/gists/ryanflorence\\"
              >Ryan Florence<!-- -->
            </a>
          </li>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/gists/mjackson\\"
              >Michael Jackson<!-- -->
            </a>
          </li>
        </ul>
      </header>
      <div data-test-id=\\"/gists/$username\\">
        <h2>All gists from mjackson</h2>
        <ul>
          <li><a>remix-server.jsx</a></li>
        </ul>
      </div>
    </div>
    <!--/$-->
    <script>
      __remixContext = {
        browserManifest: {
          \\"entry-browser\\": {
            fileName: \\"entry-browser.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
          \\"global.css\\": { fileName: \\"assets/global-ec887178.css\\" },
          \\"routes/gists\\": {
            fileName: \\"routes/gists.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
          \\"routes/gists/index\\": {
            fileName: \\"routes/gists/index.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
          \\"style/routes/gists.css\\": {
            fileName: \\"assets/style/routes/gists-d45b2a57.css\\",
          },
        },
        publicPath: \\"/build/\\",
        routeManifest: {
          \\"routes/gists\\": { id: \\"routes/gists\\", path: \\"gists\\" },
          \\"routes/gists/index\\": {
            id: \\"routes/gists/index\\",
            path: \\"/\\",
            parentId: \\"routes/gists\\",
          },
        },
        routeData: {
          \\"routes/gists\\": {
            users: [
              { id: \\"ryanflorence\\", name: \\"Ryan Florence\\" },
              { id: \\"mjackson\\", name: \\"Michael Jackson\\" },
            ],
          },
          \\"routes/gists/index\\": [
            {
              url:
                \\"https://api.github.com/gists/610613b54e5b34f8122d1ba4a3da21a9\\",
              id: \\"610613b54e5b34f8122d1ba4a3da21a9\\",
              files: { \\"remix-server.jsx\\": { filename: \\"remix-server.jsx\\" } },
              owner: {
                login: \\"ryanflorence\\",
                id: 100200,
                avatar_url:
                  \\"https://avatars0.githubusercontent.com/u/100200?v=4\\",
              },
            },
          ],
        },
        routeParams: { \\"routes/gists\\": {}, \\"routes/gists/index\\": {} },
      };
    </script>
    <script type=\\"module\\" src=\\"/build/entry-browser.js\\"></script>
  </body>
</html>
"
`;

exports[`route transitions transitioning back after a reload works 3`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <!--$-->
    <title>Public Gists</title>
    <meta name=\\"description\\" content=\\"View the latest gists from the public\\" />
    <!--/$-->
    <link
      rel=\\"stylesheet\\"
      href=\\"//unpkg.com/@exampledev/new.css@1.1.3/new.css\\"
    />
    <link rel=\\"stylesheet\\" href=\\"/build/assets/global-ec887178.css\\" />
    <link
      rel=\\"stylesheet\\"
      href=\\"/build/assets/style/routes/gists-d45b2a57.css\\"
    />
  </head>
  <body class=\\"m-4\\">
    <!--$-->
    <div data-test-id=\\"/gists\\">
      <header>
        <h1>Gists</h1>
        <ul>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/gists/ryanflorence\\"
              >Ryan Florence<!-- -->
            </a>
          </li>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/gists/mjackson\\"
              >Michael Jackson<!-- -->
            </a>
          </li>
        </ul>
      </header>
      <div data-test-id=\\"/gists/index\\">
        <h2>Public Gists</h2>
        <ul>
          <li><a>remix-server.jsx</a></li>
        </ul>
      </div>
    </div>
    <!--/$-->
    <script>
      __remixContext = {
        browserManifest: {
          \\"entry-browser\\": {
            fileName: \\"entry-browser.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
          \\"global.css\\": { fileName: \\"assets/global-ec887178.css\\" },
          \\"routes/gists\\": {
            fileName: \\"routes/gists.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
          \\"routes/gists/$username\\": {
            fileName: \\"routes/gists/$username.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
          \\"style/routes/gists.css\\": {
            fileName: \\"assets/style/routes/gists-d45b2a57.css\\",
          },
        },
        publicPath: \\"/build/\\",
        routeManifest: {
          \\"routes/gists\\": { id: \\"routes/gists\\", path: \\"gists\\" },
          \\"routes/gists/$username\\": {
            id: \\"routes/gists/$username\\",
            path: \\":username\\",
            parentId: \\"routes/gists\\",
          },
        },
        routeData: {
          \\"routes/gists\\": {
            users: [
              { id: \\"ryanflorence\\", name: \\"Ryan Florence\\" },
              { id: \\"mjackson\\", name: \\"Michael Jackson\\" },
            ],
          },
          \\"routes/gists/$username\\": [
            {
              url:
                \\"https://api.github.com/gists/610613b54e5b34f8122d1ba4a3da21a9\\",
              id: \\"610613b54e5b34f8122d1ba4a3da21a9\\",
              files: { \\"remix-server.jsx\\": { filename: \\"remix-server.jsx\\" } },
              owner: {
                login: \\"ryanflorence\\",
                id: 100200,
                avatar_url:
                  \\"https://avatars0.githubusercontent.com/u/100200?v=4\\",
              },
            },
          ],
        },
        routeParams: {
          \\"routes/gists\\": {},
          \\"routes/gists/$username\\": { username: \\"mjackson\\" },
        },
      };
    </script>
    <script type=\\"module\\" src=\\"/build/entry-browser.js\\"></script>
  </body>
</html>
"
`;

exports[`route transitions transitioning to a new route with data works 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <!--$-->
    <title>Gists Fixture App</title>
    <meta
      name=\\"description\\"
      content=\\"We're just tryin' to make sure stuff works, ya know?!\\"
    />
    <!--/$-->
    <link
      rel=\\"stylesheet\\"
      href=\\"//unpkg.com/@exampledev/new.css@1.1.3/new.css\\"
    />
    <link rel=\\"stylesheet\\" href=\\"/build/assets/global-ec887178.css\\" />
  </head>
  <body class=\\"m-4\\">
    <!--$-->
    <div data-test-id=\\"/\\">
      <header><h1>Cool Gists App</h1></header>
      <nav>
        <ul>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/gists\\">View Some gists</a>
          </li>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/user-gists/ryanflorence\\"
              >Server Redirect</a
            >
          </li>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/fart\\">Broken link</a>
          </li>
        </ul>
      </nav>
    </div>
    <!--/$-->
    <script>
      __remixContext = {
        browserManifest: {
          \\"entry-browser\\": {
            fileName: \\"entry-browser.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
          \\"global.css\\": { fileName: \\"assets/global-ec887178.css\\" },
          \\"routes/index\\": {
            fileName: \\"routes/index.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
        },
        publicPath: \\"/build/\\",
        routeManifest: { \\"routes/index\\": { id: \\"routes/index\\", path: \\"/\\" } },
        routeData: { \\"routes/index\\": null },
        routeParams: { \\"routes/index\\": {} },
      };
    </script>
    <script type=\\"module\\" src=\\"/build/entry-browser.js\\"></script>
  </body>
</html>
"
`;

exports[`route transitions transitioning to a new route with data works 2`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <!--$-->
    <title>Public Gists</title>
    <meta name=\\"description\\" content=\\"View the latest gists from the public\\" />
    <!--/$-->
    <link
      rel=\\"stylesheet\\"
      href=\\"//unpkg.com/@exampledev/new.css@1.1.3/new.css\\"
    />
    <link rel=\\"stylesheet\\" href=\\"/build/assets/global-ec887178.css\\" />
    <link
      rel=\\"stylesheet\\"
      href=\\"/build/assets/style/routes/gists-d45b2a57.css\\"
    />
  </head>
  <body class=\\"m-4\\">
    <!--$--><!--/$-->
    <div data-test-id=\\"/gists\\">
      <header>
        <h1>Gists</h1>
        <ul>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/gists/ryanflorence\\"
              >Ryan Florence
            </a>
          </li>
          <li>
            <a class=\\"text-blue-700 underline\\" href=\\"/gists/mjackson\\"
              >Michael Jackson
            </a>
          </li>
        </ul>
      </header>
      <div data-test-id=\\"/gists/index\\">
        <h2>Public Gists</h2>
        <ul>
          <li><a>remix-server.jsx</a></li>
        </ul>
      </div>
    </div>
    <script>
      __remixContext = {
        browserManifest: {
          \\"entry-browser\\": {
            fileName: \\"entry-browser.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
          \\"global.css\\": { fileName: \\"assets/global-ec887178.css\\" },
          \\"routes/index\\": {
            fileName: \\"routes/index.js\\",
            imports: [
              \\"index-d9da1d1d.js\\",
              \\"index-dcc9183b.js\\",
              \\"index-0ae7d24f.js\\",
            ],
          },
        },
        publicPath: \\"/build/\\",
        routeManifest: { \\"routes/index\\": { id: \\"routes/index\\", path: \\"/\\" } },
        routeData: { \\"routes/index\\": null },
        routeParams: { \\"routes/index\\": {} },
      };
    </script>
    <script type=\\"module\\" src=\\"/build/entry-browser.js\\"></script>
  </body>
</html>
"
`;
